### ISSUE TITLE
Tracking transformation Matrix for IE support

### ISSUE THREAD

**Issue by zero41120** [openseadragon/openseadragon#1912]
I'm working on my [PR for Path cropping](https://github.com/openseadragon/openseadragon/pull/1873)

Here is snippet from  1977 to 1985 in `tiledimage.js` in my pull request
```javascript
        var context = tiledImage._drawer._getContext(useSketch);
        var viewport  = tiledImage._drawer.viewport;
        var oldMatrix = context.getTransform(); // Save old matrix
        var scale = tiledImage.viewportToImageZoom(viewport.getZoom(true)) * $.pixelDensityRatio;
        var point = tiledImage.getOriginPixelCoordinate(true, true);
        context.translate(point.x, point.y);
        context.scale(scale, scale);
        tiledImage._drawer.clipWithPaths(tiledImage._croppingPaths, useSketch, tiledImage._croppingFillRule);
        context.setTransform(oldMatrix); // Restore old matrix
```

Currently, I'm relying on the `context.getTransform()` to provide the original transformation.
However, this feature is not supported on IE11.
I cannot use the `context.save()` and `.restore()`, because it will undo the cropping.
(Or maybe I just don't understand the system enough)

Base on this [StackOverflow question](https://stackoverflow.com/questions/7395813/html5-canvas-get-transform-matrix) asked almost 10 years ago. It doesn't seem like there is any way that IE11 can provide the transformation matrix from the `canvas` object.

All the answers suggest some form of tracking matrix change by the application.
I think it will be really beneficial to track the transformation matrix.

Maybe if someone from OSD who's really familiar with the system can wrap the context and track this information within OSD, or can give me the correct direction to implement this wrapper.

**Comment by iangilman** [comment]
Good point! That said, since the current need is for https://github.com/openseadragon/openseadragon/pull/1873 which adds a feature that relies on Path2D, another thing that IE 11 doesn't support, I think it'll be OK… We just need to make sure the entire feature fails gracefully (i.e. doesn't do anything) on IE 11.

It's good to have this issue noted, however, in case we end up wanting something like this for additional features.
